src/components/Waveform/Waveform.tsx

```diff
@@ -129,7 +132,10 @@ export const Waveform = forwardRef<IWaveformRef, IWaveform>((props, ref) => {
     try {
       const prepare = await preparePlayerForPath();
       if (prepare) {
-        await getAudioDuration();
+        const duration = await getAudioDuration();
+        if (duration < 0) {
+          await getAudioDuration();
```

Comment:
@nilesh-simform  Why do we need to get Duration again and for second time.

Reply:
Sometimes it returns a negative value, indicating that the player is not prepared or due to some other issue. Therefore, when a negative value is received, we call getAudioDuration again to retrieve the audio duration (this works for cases where we were unable to provide the duration because the player was not prepared the first time)